cli/bpmetadata/validate.go

```diff
@@ -0,0 +1,128 @@
+package bpmetadata
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/util"
+	"github.com/xeipuuv/gojsonschema"
+	"sigs.k8s.io/yaml"
+)
+
+//go:embed schema/bpmetadataschema.json
+var s []byte
+
+// validateMetadata validates the metadata files for the provided
+// blueprint path. This validation occurs for top-level blueprint
+// metadata and blueprints in the modules/ folder, if present
+func validateMetadata(bpPath, wdPath string) error {
+	// load schema from the binary
+	schemaLoader := gojsonschema.NewStringLoader(string(s))
+
+	// check if the provided output path is relative
+	if !path.IsAbs(bpPath) {
+		bpPath = path.Join(wdPath, bpPath)
+	}
+
+	var vErrs []error
+	err := validateMetadataYaml(bpPath, schemaLoader)
+	if err != nil {
+		vErrs = append(vErrs, err)
+		Log.Error("metadata validation failed", "err", err)
+	}
```

Comment:
nit: it would keep the code more left aligned if we do the path collection first i.e `bpPath` + any module path and then in the next block iterate and do `validateMetadataYaml` over all collected paths.

Reply:
Done.